public/app/features/query/state/PanelQueryRunner.ts

```diff
@@ -105,97 +108,103 @@ export class PanelQueryRunner {
     }
 
     return this.subject.pipe(
-      this.getTransformationsStream(withTransforms),
-      map((data: PanelData) => {
-        let processedData = data;
-        let streamingPacketWithSameSchema = false;
+      mergeMap((data: PanelData) => {
+        let fieldConfig = this.dataConfigSource.getFieldOverrideOptions();
 
-        if (withFieldConfig && data.series?.length) {
-          if (lastConfigRev === this.dataConfigSource.configRev) {
-            const streamingDataFrame = data.series.find((data) => isStreamingDataFrame(data)) as
-              | StreamingDataFrame
-              | undefined;
+        if (data.series === lastRawFrames && lastFieldConfig?.fieldConfig === fieldConfig?.fieldConfig) {
+          return of({ ...data, structureRev, series: lastProcessedFrames });
```

Comment:
This is the main performance improvement correct? Early return of same data when data and field config haven't changed since last cycle?

Reply:
the issue is that visualizations will re-render/process data when it gets a new instance.  When we issue a new query, the current mechanics resend the same data, but not in the same *instance* so this forces a re-render/process when the query is requested and again when we receive the new data

Reply:
By `new instance` do we mean new object in memory? and ok - so this is preventing unnecessary renders of panels by us sending the `series: lastProcessedFrames` which is the same object in memory as before - hence the changes to viz panels looking at `data.series` vs `data` in memo hooks

Reply:
yes